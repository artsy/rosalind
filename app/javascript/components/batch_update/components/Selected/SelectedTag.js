import React from 'react'
import SelectedComponent from './SelectedComponent'

const SelectedTag = (props) => {
  return (
    <SelectedComponent
      name={props.name}
      onRemove={props.onRemove}
    />
  )
}

export { SelectedTag }
