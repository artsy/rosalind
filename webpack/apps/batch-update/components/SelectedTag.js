import React from 'react'

class SelectedTag extends React.Component {
  constructor (props) {
    super(props)
    this.handleRemove = this.handleRemove.bind(this)
  }

  handleRemove (e) {
    e.preventDefault()
    this.props.onRemoveTag(this.props.name)
  }

  render () {
    const { className, name } = this.props
    return (
      <div className={className}>
        {name}
        <a href='#' className='remove'
          onClick={this.handleRemove}>âœ•</a>
      </div>
    )
  }
}

SelectedTag.propTypes = {
  name: React.PropTypes.string.isRequired,
  onRemoveTag: React.PropTypes.func.isRequired
}

/* default styled component */

import styled from 'styled-components'

const StyledSelectedTag = styled(SelectedTag)`
  font-weight: bold;
  margin: 0.25em 0;

  a.remove {
    display: inline-block;
    height: 2em;
    width: 2em;
    line-height: 2em;
    text-align: center;
    color: #ddd;
    margin-left: 0.5em;
    border-radius: 1em;
    text-decoration: none;
    &:hover {
      color: #333;
      cursor: pointer;
      background: #eee;
    }
  }
`

export default StyledSelectedTag
