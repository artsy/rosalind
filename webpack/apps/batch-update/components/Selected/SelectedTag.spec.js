import React from 'react'
import renderer from 'react-test-renderer'
import { SelectedTag } from './SelectedTag'

let props

beforeEach(() => {
  props = {
    name: 'Clown',
    onRemove: jest.fn()
  }
})

it('renders correctly', () => {
  const rendered = renderer.create(<SelectedTag {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})
