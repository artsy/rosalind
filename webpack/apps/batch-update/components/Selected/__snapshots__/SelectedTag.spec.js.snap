// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="crrNnH"
>
  Clown
  <a
    className="remove"
    href="#"
    onClick={[Function]}
  >
    âœ•
  </a>
</div>
`;
